src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,76 @@
-export const Pagination = () => {};
+import React from 'react';
+
+interface PaginationProps {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<PaginationProps> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const handlePageChange = (page: number) => {
+    if (page >= 1 && page <= totalPages) {
```

Comment:
The condition `page >= 1 && page <= totalPages` is a bit risky. If for any reason the currentPage becomes 0 or less, or more than totalPages, the user would be stuck and can't navigate to other pages. It might be better to handle these edge cases and ensure the page is always within valid range.